packages/utils/src/locking.js

```diff
@@ -59,8 +76,27 @@ export async function lock({ id, dir, root, options }) {
   lockPaths.add(lockPath)
 
   return async function lockRelease() {
-    await release()
-    await fs.remove(lockPath)
-    lockPaths.delete(lockPath)
+    try {
+      await fs.remove(lockPath)
+      lockPaths.delete(lockPath)
+
+      // release as last so the lockPath is still removed
+      // when it fails on a compromised lock
+      await release()
+    } catch (e) {
+      if (lockWasCompromised && e.message.includes('already released')) {
+        // proper-lockfile doesnt remove lockDir when lock is compromised
+        // removing it here could cause the 'other' process to throw an error
+        // as well, but in our case its much more likely the lock was
+        // compromised due to mtime update timeouts
+        const lockDir = `${lockPath}.lock`
+        if (await fs.exists(lockDir)) {
+          await fs.remove(lockDir)
+        }
+        return
+      }
+
+      consola.warn(e)
```

Comment:
Can we early return by checking opposite of if condition?

Reply:
Also what about using debug level?

Reply:
Thats probably a good idea indeed here